lib/jake/string.ex

```diff
@@ -1,21 +1,46 @@
 defmodule Jake.String do
-  def gen(spec) do
-    options = []
-
-    options =
-      if min_length = Map.get(spec, "minLength") do
-        Keyword.put(options, :min_length, min_length)
-      else
-        options
-      end
-
-    options =
-      if max_length = Map.get(spec, "maxLength") do
-        Keyword.put(options, :max_length, max_length)
-      else
-        options
-      end
-
-    StreamData.string(:ascii, options)
+  @strlen_min 1
+
+  @strlen_max 100
+
+  def gen_string(map), do: stringer(map, map["enum"], map["pattern"])
+
+  def find_min_max(map) do
+    min = Map.get(map, "minLength", @strlen_min)
+    max = Map.get(map, "maxLength", @strlen_max)
+    {min, max}
+  end
+
+  def stringer(map, enum, pattern) when is_nil(enum) and is_nil(pattern) do
+    {min, max} = find_min_max(map)
+    re = Randex.stream(~r/[a-zA-Z0-9\_]{#{min},#{max}}/)
+
+    if min <= max do
+      StreamData.bind_filter(StreamData.string(:alphanumeric), fn
+        x when byte_size(x) in min..max ->
+          {:cont, StreamData.constant(x)}
+
+        x when true ->
+          {:cont, Enum.take(re, 1) |> StreamData.member_of()}
```

Comment:
This will return the same value every time?

Reply:
No, when I checked, it was returning different value every time.

Reply:
ok, but it's not supposed to be used like this. you could just use `Randex.stream(~r/[a-zA-Z0-9\_]{#{min},#{max}}/, mod: Randex.Generator.StreamData)`, because randex itself just returns a generator

Reply:
> it's not supposed to be used like this

I've also realised it
In the beginning, I was somewhat confused about how to convert Randex.stream into StreamData. Later I came to know about it when I looked at **activesphere/jake** implementation. But as I said in the previous post, I wanted to know your opinion on it.
I'll change the method.